analyzers/src/SonarAnalyzer.CSharp/Rules/Utilities/TokenTypeAnalyzer.cs

```diff
@@ -171,6 +172,10 @@ private bool IsValueParameterOfSetter(SimpleNameSyntax simpleName)
                         }
                     };
 
+            private static bool IsLeftMostMemberAccess(SimpleNameSyntax simpleName)
+                => simpleName is { Parent: not MemberAccessExpressionSyntax }
+                    || (simpleName is { Parent: MemberAccessExpressionSyntax { Expression: { } expression } } && expression == simpleName);
```

Comment:
Since the parent is always `MemberAccessExpressionSyntax` we can use a cast instead and simplify the check.
```suggestion
                    || ((MemberAccessExpressionSyntax)simpleName.Parent is { Expression: { } expression } && expression == simpleName);
```

Reply:
How is this simpler? You can no longer rearrange the conditions without causing trouble.

Reply:
We don't need to rearrange the conditions. Also, I don't see a point in verifying a condition twice.

Reply:
The code seems similar with:
```
if (a)
else if (!a && b)
```
on the else branch, we know already that `a` is false.

Reply:
I need to cast to MemberAccessExpressionSyntax to get access to the Expression property. This is not redundant as described in your example. We do not save anything here and it would be the only time we use a cast in the whole file (except the unavoidable wrapper casts). Please approve it without the change.

Reply:
`simpleName is { Parent: MemberAccessExpressionSyntax` this is more than a cast, it's a condition. A condition that we know it's always true. I find it redundant and confusing.

Reply:
The PR is already approved, you can merge it if you want but I still think that using a direct cast describes better the intent and makes the code clearer.

Reply:
.. and it get's optimized away in release builds anyway:
https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE+UkALIgLIAUAogB4DGApgA4AuAlgPYB2itliAlJtlxCAbgEMATonG0AzgFcANo0QBeLjxkB3Zo2oALQUJxYYRs4natlAQXEBzOQFta7RlTpM2nFQD5EjcTlaABpDcxxbB2dXdwYWDkxEAEl2dlpxWM8OAC5ESKcXNxo4rwBfVT8AM1EFGRCw8IB9CsRq2vrTc1KAbgbcEAgATlJpeSVeXs6cUthSoA=

Reply:
My point was not about being optimize or not 😃

Reply:
if you choose to won't fix, can you document this in SC? I understand this si related to the bug in SC? https://sonarcloud.io/project/issues?resolved=false&types=BUG&pullRequest=7822&id=sonaranalyzer-dotnet&open=AYoIBGMsGbABB2uAfdJd

Reply:
Done

Reply:
The raised bug seems to be wrong. There are multiple conditions there and it's hard to understand which one is actually always false.

One part of the condition (`simpleName is { Parent: MemberAccessExpressionSyntax`) is actually always true since we check exactly the opposite before the `||`